### PR TITLE
fix(argo-cd): Update to Argo CD v2.3.4

### PR DESCRIPTION
**Pull request by crenshaw-dev**
Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [x] I have bumped the chart version according to [versioning](https://github.com/argoproj/argo-helm/blob/master/CONTRIBUTING.md#versioning)
* [x] I have updated the documentation according to [documentation](https://github.com/argoproj/argo-helm/blob/master/CONTRIBUTING.md#documentation)
* [x] I have updated the chart changelog with all the changes that come with this pull request according to [changelog](https://github.com/argoproj/argo-helm/blob/master/CONTRIBUTING.md#changelog).
* [x] Any new values are backwards compatible and/or have sensible default.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo).
* [ ] My build is green ([troubleshooting builds](https://argoproj.github.io/argo-cd/developer-guide/ci/)).

Changes are automatically published when merged to `master`. They are not published on branches.

### CODE REVIEW

**Comment by crenshaw-dev** [comment]
Not sure how to fix the lint error. I ran `bash scripts/helm-docs.sh`, and no changes were made.

**Review by mkilchhofer — APPROVED** [review]

File: charts/argo-cd/README.md [hunk 1]
```diff
@@ -91,8 +91,8 @@ Please use `kubectl` to upgrade CRDs manually from [crds](crds/) folder or via t
 ```bash
 kubectl apply -k https://github.com/argoproj/argo-cd.git/manifests/crds?ref=<appVersion>
 
-# Eg. version v2.3.3
-kubectl apply -k https://github.com/argoproj/argo-cd.git/manifests/crds?ref=v2.3.3
+# Eg. version v2.3.4
+kubectl apply -k https://github.com/argoproj/argo-cd.git/manifests/crds?ref=v2.3.4
```
Comment (tadeboro) [inline charts/argo-cd/README.md:95]: I am not 100% sure, but I think we need to make those changes in the README template and then render the the actual README markdown file.

**Review by tadeboro — CHANGES_REQUESTED** [review]

File: charts/argo-cd/README.md (hunk 1 above)
Comment (mkilchhofer) [inline charts/argo-cd/README.md:95]:
```suggestion
# Eg. version v2.3.3
kubectl apply -k https://github.com/argoproj/argo-cd.git/manifests/crds?ref=v2.3.3
```

It's just an example, lets revert this :-)

**Review by mkilchhofer — APPROVED** [review]

File: charts/argo-cd/README.md (hunk 1 above)
Comment (mkilchhofer) [reply charts/argo-cd/README.md]: As it's just an example we can easily revert this change. Then the CI pipeline is happy :-)

**Pull request merged**